src/command_modules/azure-cli-search/azure/cli/command_modules/search/commands.py

```diff
@@ -0,0 +1,44 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+# pylint: disable=line-too-long
+
+
+def load_command_table(self, _):
+    from azure.cli.core.commands import CliCommandType
+    from azure.cli.command_modules.search._client_factory import cf_search_services, cf_search_admin_keys, \
+        cf_search_query_keys
+
+    search_services_sdk = CliCommandType(
+        operations_tmpl='azure.mgmt.search.operations#ServicesOperations.{}',
+        client_factory=cf_search_services
+    )
+
+    search_admin_keys_sdk = CliCommandType(
+        operations_tmpl='azure.mgmt.search.operations#AdminKeysOperations.{}',
+        client_factory=cf_search_admin_keys
+    )
+
+    search_query_keys_sdk = CliCommandType(
+        operations_tmpl='azure.mgmt.search.operations#QueryKeysOperations.{}',
+        client_factory=cf_search_query_keys
+    )
+
+    with self.command_group('search services', search_services_sdk) as g:
+        # right now list_by_resource_group is the only way to list, so directly map to list_by_resource_group.
+        g.command('list', 'list_by_resource_group')
+        g.command('show', 'get')
```

Comment:
This should be a call to `g.show_command('show')`

Reply:
@tjprescott I guess you mean g.generic_show_command('show'). I tried that but it cannot raise exceptions accordingly.
I searched entire repo and only find one place using generic_show_command. Thus cannot find a solution. 

```
self.cmd('az search services delete -n {name} -g {rg}')
with self.assertRaises(CloudError) as err:
    self.cmd('az search services show -n {name} -g {rg}')
```

Reply:
It should just be `g.show_command(…)`  You might need to rebase on the latest dev if you are seeing `generic_show_command`.  cc/ @williexu

Reply:
@tjprescott okay, let me try...

Reply:
Yes, show_command is there after rebase. But still it wont throw CloudError as expected.
I modified my test anyway.